src/cluster_legacy.c

```diff
@@ -2815,10 +2874,52 @@ static clusterNode *getNodeFromLinkAndMsg(clusterLink *link, clusterMsg *hdr) {
     return sender;
 }
 
+void processPublishPacket(clusterMsgDataPublish *publish_data, uint16_t type) {
+    robj *channel, *message;
+    uint32_t channel_len, message_len;
+
+    /* Don't bother creating useless objects if there are no
+     * Pub/Sub subscribers. */
+    if ((type == CLUSTERMSG_TYPE_PUBLISH && serverPubsubSubscriptionCount() > 0) ||
+        (type == CLUSTERMSG_TYPE_PUBLISHSHARD && serverPubsubShardSubscriptionCount() > 0)) {
+        channel_len = ntohl(publish_data->channel_len);
+        message_len = ntohl(publish_data->message_len);
+        channel = createStringObject((char *)publish_data->bulk_data, channel_len);
+        message = createStringObject((char *)publish_data->bulk_data + channel_len, message_len);
+        pubsubPublishMessage(channel, message, type == CLUSTERMSG_TYPE_PUBLISHSHARD);
+        decrRefCount(channel);
+        decrRefCount(message);
+    }
+}
+
+void processLightPacket(clusterLink *link, uint16_t type) {
+    clusterMsgLight *hdr = (clusterMsgLight *)link->rcvbuf;
+
+    if (type == CLUSTERMSG_TYPE_PUBLISH || type == CLUSTERMSG_TYPE_PUBLISHSHARD) {
+        processPublishPacket(&hdr->data.publish.msg, type);
+    }
+}
+
+static inline int messageTypeSupportsLightHdr(uint16_t type) {
+    switch (type) {
+    case CLUSTERMSG_TYPE_PUBLISH: return 1;
+    case CLUSTERMSG_TYPE_PUBLISHSHARD: return 1;
+    }
+    return 0;
+}
+
+
 int clusterIsValidPacket(clusterLink *link) {
     clusterMsg *hdr = (clusterMsg *)link->rcvbuf;
     uint32_t totlen = ntohl(hdr->totlen);
-    uint16_t type = ntohs(hdr->type);
+    int is_light = IS_LIGHT_MESSAGE(ntohs(hdr->type));
+    uint16_t type = ntohs(hdr->type) & ~CLUSTERMSG_MODIFIER_MASK;
+
+    if (is_light && !messageTypeSupportsLightHdr(type)) {
+        serverLog(LL_NOTICE, "--- Packet of type %s does not support light cluster header",
+                  clusterGetMessageTypeString(type));
```

Comment:
I don't think we use "--- " as a prefix for NOTICE log messages? Also let's print out the numeric type value too in case a new unknown type is introduced in the future version, which is unknown to this version. 
```suggestion
        serverLog(LL_NOTICE, "Packet of type '%s' (%u) does not support light cluster header. Marking packet as invalid.",
                  clusterGetMessageTypeString(base_type), base_type);

```